cli/src/package.rs

```diff
@@ -1,36 +1,136 @@
-use std::{env::current_dir, fs, path::PathBuf};
-
+use crate::error::CliError;
 use directories::ProjectDirs;
+use futures::future::join_all;
+use modmark_core::Resolve;
+use serde_json;
+use std::{
+    env::current_dir,
+    fs::{self, create_dir_all, File},
+    io::copy,
+    path::PathBuf,
+};
 
-pub trait Resolve {
-    fn resolve(&self, path: &str) -> Option<Vec<u8>>;
-}
+pub struct PackageManager {}
 
-pub struct PackageManager;
 impl Resolve for PackageManager {
-    fn resolve(&self, path: &str) -> Option<Vec<u8>> {
-        let mut local_path = match current_dir() {
-            Ok(p) => p,
-            Err(_) => return None,
-        };
+    type Error = CliError;
+    fn resolve(&self, path: &str) -> Result<Vec<u8>, Self::Error> {
+        let out = tokio::runtime::Builder::new_multi_thread()
```

Comment:
We might want to make the whole package manager into a struct which we use as a singleton, which starts a tokio runtime the first time we actually run it, so we don't make new runtimes all the time (which actually creates new OS threads, one per CPU core etc). Also, we might not need to spin up a mutli-thread runtime if we only want to resolve one package, since yeah...it is only one job. Maybe add `.worker_threads(1)`? Or, if we have a singleton which owns its runtime, we may just schedule it there..